### PR TITLE
Preserve sub-second data for time scalars in column construction

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Fixes: #15654 

This PR makes fixes such that sub-second timestamp data is not being dropped in column construction.

Forks out of https://github.com/rapidsai/cudf/pull/14534/
## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/_lib/scalar.pyx [hunk 1]
```diff
@@ -354,7 +354,8 @@ def as_device_scalar(val, dtype=None):
 def _is_null_host_scalar(slr):
     if cudf.utils.utils.is_na_like(slr):
         return True
-    elif isinstance(slr, (np.datetime64, np.timedelta64)) and np.isnat(slr):
+    elif (isinstance(slr, (np.datetime64, np.timedelta64)) and np.isnat(slr)) or \
+            isinstance(slr, type(pd.NaT)):
```
Comment (mroeschke) [inline python/cudf/cudf/_lib/scalar.pyx:358]: Nit: Could you use `slr is pd.NaT` (and below)?

File: python/cudf/cudf/core/column/column.py [hunk 2]
```diff
@@ -2163,6 +2163,15 @@ def as_column(
                     nan_as_null=nan_as_null,
                     length=length,
                 )
+            elif isinstance(
+                element, (pd.Timestamp, pd.Timedelta, type(pd.NaT))
+            ):
```
[the remark this answers is not in the document]
Comment (galipremsagar) [reply python/cudf/cudf/core/column/column.py:2168]: > Could you reference [apache/arrow#26492](https://github.com/apache/arrow/issues/26492) here?

Done



> And add the `pd.NaT` to the `elif not any(element is na for na in (None, pd.NA, np.nan)):` below?

`pd.NaT` will be handled by elif in `2166` no?

File: python/cudf/cudf/_lib/scalar.pyx (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/_lib/scalar.pyx]: Done

File: python/cudf/cudf/core/column/column.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mroeschke) [reply python/cudf/cudf/core/column/column.py:2169]: > pd.NaT will be handled by elif in 2166 no?

Ah actually I think how you have it already might be better. We generally would want `pd.NaT` to be handled by pandas and not accidentally get parsed by pyarrow I think

**Review by mroeschke — APPROVED** [review]

**Pull request merged**